### DISCUSSION TITLE
Enforcing revisions policy fails with null reference exception on deleted documents

### DISCUSSION THREAD

**Discussion by bwmeyer** [ravendb/ravendb#17306] [General]
I tried to enforce revisions policy on a database via Studio.
Unfortunately, the result is an NullReferenceExcpetion:

<img width="676" alt="image" src="https://github.com/ravendb/ravendb/assets/39671001/807a4afe-26ac-46f9-83b0-98660bae874e">

I tried to debug with dnspy.
It seems, that the error is happening only on documents which are deleted an kept in the revisions bin.

<img width="1133" alt="image" src="https://github.com/ravendb/ravendb/assets/39671001/aef380d7-0d1c-4792-9331-59ad76d27771">

Edit:
It looks like the id on which the exception is throwing, is not found by the system at all.
If I enter the id in the Studios "go to document" to open the document directly and press enter, just the first entry of the recent document pull down will be opened.
The same happens if I enter an id which is invalid like "sahdgfkjahdsgfas".
I would assume to get an error page about "not found id 'sahdgfkjahdsgfas'"

**Comment by shaharhikri** [comment]
We opened an issue for that.
You can check its progress here:
https://issues.hibernatingrhinos.com/issue/RavenDB-21382/NRE-in-enforce-revisions-configuration
